### DISCUSSION TITLE
Adjustable Quick Ranges/Time Ranges

### DISCUSSION THREAD

**Discussion by BlackDex** [grafana/grafana#71847] [General]
Hello there,

I wonder if it is already possible, and if not, if it could be added to have an adjustable or configurable `Quick Ranges`/`Time Range` picker.

Currently there are a lot of defaults set `Last 7 days`, `Last 20 days`, `Last 1 year`, `Last 2 years` etc...
Since we have a very huge backend of datapoints, and some users just select `Last 1 year` for example, and then type a Prometheus Query for example, it is going to try and autocomplete the labels. This causes our services to overload and in the end crash.

Adding more pods or resources will not help, that will just be to much.

If i was able to somehow limit that whole list, or adjust what is visible for all the users, that might help in preventing these issues, and people really need to think about selecting a larger range them self.

It would also be nice for setups where the retention only is configured for something like 3 months or so, having an option to select a whole year just doesn't seem to be logical to be visible in these cases.

So, i see multiple use-case's for something like this.

Again, if this already is possible, please educate me on how, else i would request for this to be a feature.

Thanks!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82850. Feel free to continue the discussion around this there. Thank you!
